File: src/LEContagem/app/src/main/java/com/example/lecontagem/CadastroActivity.java
package com.example.lecontagem;

import android.content.Intent;
import android.os.Bundle;
import android.util.Patterns;
import android.view.View;
import android.widget.ArrayAdapter;
import android.widget.Button;
import android.widget.Spinner;
import android.widget.TextView;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

import com.google.android.material.textfield.TextInputEditText;

public class CadastroActivity extends AppCompatActivity {

    TextInputEditText inputNome;
    TextInputEditText inputEmail;
    TextInputEditText inputSenha;

    Spinner spinnerCargo;

    Button cadastrarButton;

    TextView voltarLogin;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.cadastro);

        inputNome = findViewById(R.id.InputNome);
        inputEmail = findViewById(R.id.InputEmail);
        inputSenha = findViewById(R.id.InputSenha);

        spinnerCargo = findViewById(R.id.spinnerCargo);

        cadastrarButton = findViewById(R.id.CadastrarButton);
        voltarLogin = findViewById(R.id.VoltarLogin);

        String[] cargos = {
                "Selecione o cargo",
                "Funcionário",
                "Nutricionista",
                "Gestor",
                "Administrador"
        };

        ArrayAdapter<String> adapter = new ArrayAdapter<>(
                this,
                android.R.layout.simple_spinner_dropdown_item,
                cargos
        );

        spinnerCargo.setAdapter(adapter);

        cadastrarButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {

                String nome = inputNome.getText().toString().trim();
                String email = inputEmail.getText().toString().trim();
                String senha = inputSenha.getText().toString().trim();
                String cargo = spinnerCargo.getSelectedItem().toString();

                if (nome.isEmpty()) {
                    inputNome.setError("Digite seu nome");
                    inputNome.requestFocus();
                    return;
                }

                if (email.isEmpty()) {
                    inputEmail.setError("Digite seu email");
                    inputEmail.requestFocus();
                    return;
                }

                if (!Patterns.EMAIL_ADDRESS.matcher(email).matches()) {
                    inputEmail.setError("Email inválido");
                    inputEmail.requestFocus();
                    return;
                }

                if (senha.isEmpty()) {
                    inputSenha.setError("Digite sua senha");
                    inputSenha.requestFocus();
                    return;
                }

                if (cargo.equals("Selecione o cargo")) {
                    Toast.makeText(CadastroActivity.this, "Selecione um cargo", Toast.LENGTH_SHORT).show();
                    return;
                }

                Toast.makeText(CadastroActivity.this, "Cadastro realizado com sucesso!", Toast.LENGTH_SHORT).show();

                Intent intent = new Intent(CadastroActivity.this, LoginActivity.class);

                intent.putExtra("email", email);
                intent.putExtra("senha", senha);

                startActivity(intent);
                finish();
            }
        });

        voltarLogin.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {

                Intent intent = new Intent(CadastroActivity.this, LoginActivity.class);
                startActivity(intent);
                finish();
            }
        });
    }
}

File: src/Entregas/frontend/LEContagem/app/src/main/java/com/example/lecontagem/CadastroActivity.kt
package com.example.lecontagem

import android.os.Bundle
import android.widget.*
import androidx.appcompat.app.AppCompatActivity
import com.google.android.material.textfield.TextInputEditText
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.firestore.FirebaseFirestore

class CadastroActivity : AppCompatActivity() {

    private val auth = FirebaseAuth.getInstance()
    private val firestore = FirebaseFirestore.getInstance()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.cadastro)

        val btnCadastrar = findViewById<Button>(R.id.CadastrarButton)
        val spinnerCargo = findViewById<Spinner>(R.id.spinnerCargo)

        val cargos = arrayOf("Selecione o cargo", "Operador", "Coordenação", "Admin")
        spinnerCargo.adapter = ArrayAdapter(this, android.R.layout.simple_spinner_dropdown_item, cargos)

        btnCadastrar.setOnClickListener {
            val email = findViewById<TextInputEditText>(R.id.InputEmail).text.toString()
            val senha = findViewById<TextInputEditText>(R.id.InputSenha).text.toString()
            val nome = findViewById<TextInputEditText>(R.id.InputNome).text.toString()
            val cargo = spinnerCargo.selectedItem.toString()

            if (email.isNotEmpty() && cargo != "Selecione o cargo") {
                auth.createUserWithEmailAndPassword(email, senha).addOnSuccessListener {
                    val uid = it.user?.uid ?: ""
                    val map = hashMapOf("uid" to uid, "nome" to nome, "cargo" to cargo, "email" to email)
                    firestore.collection("users").document(uid).set(map).addOnSuccessListener {
                        Toast.makeText(this, "Sucesso!", Toast.LENGTH_SHORT).show()
                        finish()
                    }
                }
            }
        }
    }
}